Composer/packages/client/src/pages/knowledge-base/QnAPage.tsx

```diff
@@ -34,10 +51,26 @@ const QnAPage: React.FC<RouteComponentProps<{
   const actions = useRecoilValue(dispatcherState);
   const dialogs = useRecoilValue(dialogIdsState(actualProjectId));
   const qnaFiles = useRecoilValue(qnaFilesState(actualProjectId));
-  //To do: support other languages
-  const locale = 'en-us';
-  //const locale = useRecoilValue(localeState);
-  const creatQnAOnInfo = useRecoilValue(createQnAOnState);
+  const locale = useRecoilValue(localeState(actualProjectId));
+  const createQnAOnInfo = useRecoilValue(createQnAOnState);
+  const settings = useRecoilValue(settingsState(actualProjectId));
+  const { defaultLanguage } = settings;
+  const languages = useMemo(() => (defaultLanguage === locale ? [defaultLanguage] : [locale, defaultLanguage]), [
+    locale,
+    defaultLanguage,
+  ]);
+  const [currentLocale, setCurrentLocale] = useState(locale);
+
+  const showTabBar = useMemo(() => {
+    const targetFileId = dialogId.endsWith(qnaSuffix(locale)) ? dialogId : `${dialogId}.${locale}`;
+    const qnaFile = qnaFileId
+      ? qnaFiles.find(({ id }) => id === qnaFileId)
+      : qnaFiles.find(({ id }) => id === targetFileId);
+    return !(defaultLanguage === locale || qnaFile?.empty);
+  }, [defaultLanguage, locale, qnaFiles, dialogId, qnaFileId]);
+  useEffect(() => {
+    setCurrentLocale(locale);
+  }, [locale]);
```

Comment:
This effect hook looks like an anti-pattern usage

Reply:
use 
```
  useEffect(() => {
    if (locale !== currentLocale) {
      setCurrentLocale(locale);
    }
  }, [locale]);
```